eigensync/src/database.rs

```diff
@@ -0,0 +1,84 @@
+use std::{collections::HashMap, path::PathBuf};
+
+use anyhow::{Context, Ok, Result};
+
+use automerge::Change;
+use libp2p::PeerId;
+use sqlx::SqlitePool;
+use tracing::info;
+
+use crate::protocol::SignedEncryptedSerializedChange;
+
+#[derive(Clone)]
+pub struct Database {
+    pub pool: SqlitePool,
+}
+
+impl Database {
+    pub async fn new(data_dir: PathBuf) -> Result<Self> {
+        if !data_dir.exists() {
+            std::fs::create_dir_all(&data_dir)?;
+            info!("Created server database directory: {}", data_dir.display());
+        }
+
+        let db_path = data_dir.join("changes");
+        let database_url = format!("sqlite:{}?mode=rwc", db_path.display());
+        let pool = SqlitePool::connect(&database_url).await?;
+
+        let db = Self { pool };
+        db.migrate().await?;
+
+        Ok(db)
+    }
+
+    async fn migrate(&self) -> Result<()> {
+        sqlx::migrate!("./migrations").run(&self.pool).await?;
+        info!("Server changes database migration completed");
+        Ok(())
+    }
+
+    pub async fn get_peer_changes(&self, peer_id: PeerId) -> Result<Vec<SignedEncryptedSerializedChange>> {
+        let peer_id = peer_id.to_string();
+        
+        let rows = sqlx::query!(
+            r#"
+            SELECT change
+            FROM change
+            WHERE peer_id = ?
+            ORDER BY id DESC
+            "#,
+            peer_id
+        )
+        .fetch_all(&self.pool)
+        .await?;
+
+        let mut changes = Vec::new();
```

Comment:
You should be able to use `.map(...)` here

Something like `rows.iter().map(|row| SignedEncryptedSerializedChange::new(row.change.clone))